src/mongoc/mongoc-topology.c

```diff
@@ -351,29 +359,23 @@ mongoc_topology_select (mongoc_topology_t         *topology,
          topology->stale = false;
       }
 
-      /* until we find a server or timeout */
-      for (;;) {
```

Comment:
Oh, wait, nevermind. We have to keep the "for (;;)" loop in. Don't change this code until / unless the SS spec changes. We're still implementing this:

https://github.com/mongodb/specifications/blob/master/source/server-selection/server-selection.rst#single-threaded-server-selection-implementation

In the future, an expire_at value of -42 or something may have special meaning, but not yet.